### PR TITLE
Network slimming

### PR DESCRIPTION
**Pull request by Nahush26**
Have doubt in dataloader part , next step is to create the experiment notebook.

### CODE REVIEW

**Review by dkgupta90 — CHANGES_REQUESTED** [review]
@Nahush26 please also add a working notebook (clean and commented) that demonstrates the process of network slimming on cifar10. See the chipnet notebook from @rishabh-16. It is still to be cleaned, but try making your a clean one.

File: trailmet/algorithms/prune/network_slimming_prune.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
+import os
```
Comment (dkgupta90) [inline trailmet/algorithms/prune/network_slimming_prune.py]: there should only be a single file network_slimming.py and it should have everything in a modular manner organized into classes.

File: trailmet/algorithms/prune/network_slimming_prune.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+class get_params():
+  def __init__(self,dataset,test_bs,percent,model,save,depth,arch):
+    self.dataset=dataset
+    self.test_bs=test_bs
+    self.percent=percent  # percentage of network to be pruned
+    self.model=model      # path where the model to be pruned is saved
+    self.save=save        # path where pruned model will be saved
+    self.depth=depth      # depth of model if arch is resnet
+    self.arch=arch        # vgg-16 and resnet family is supported
+initial=get_params('cifar10',64,0.6,'/content/drive/MyDrive/VGG-16 NS/CIFAR-10/trained/resnet_model_best.pth.tar','/content/drive/MyDrive/VGG-16 NS/CIFAR-10/pruned/resnet_pruned1.pth.tar',164,'resnet')
```
Comment (dkgupta90) [inline trailmet/algorithms/prune/network_slimming_prune.py]: all hard coded parameters should be kept in a yaml file.

File: trailmet/algorithms/prune/network_slimming_prune.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+""
+"""
+defining model architecture
+"""
+class prune_it:
```
Comment (dkgupta90) [inline trailmet/algorithms/prune/network_slimming_prune.py]: this is not really the format. prune_it seems more of a function and should not be a class. Follow the structure in chipnet.py. Your code is the closest to chipnet and it should be easy to organize accordingly

File: trailmet/algorithms/prune/network_slimming_prune.py [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+                                if layer_id_in_cfg < len(cfg_mask):
+                                    end_mask = cfg_mask[layer_id_in_cfg]
+                            else:
+                                m1.weight.data = m0.weight.data[idx1.tolist()].clone()
+                                m1.bias.data = m0.bias.data[idx1.tolist()].clone()
+                                m1.running_mean = m0.running_mean[idx1.tolist()].clone()
+                                m1.running_var = m0.running_var[idx1.tolist()].clone()
+                                layer_id_in_cfg += 1
+                                start_mask = end_mask.clone()
+                                if layer_id_in_cfg < len(cfg_mask):  # do not change in Final FC
```
Comment (dkgupta90) [inline trailmet/algorithms/prune/network_slimming_prune.py]: there are a lot of parts of this code which can be made more structured. see chipnet.py and follow similar guidelines.

File: trailmet/algorithms/prune/network_slimming_train.py [hunk 5]
```diff
@@ -0,0 +1,253 @@
+from __future__ import print_function
```
Comment (dkgupta90) [inline trailmet/algorithms/prune/network_slimming_train.py]: i am skipping review of this for now. There should be a singile file with organized content.

File: trailmet/models/channel_selection_ns.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import numpy as np
+import torch
+import torch.nn as nn
+
+
+class channel_selection(nn.Module):
```
Comment (dkgupta90) [inline trailmet/models/channel_selection_ns.py]: why do you have one class in one file - this is not right. see ChipNet.

**Review by dkgupta90 — COMMENTED** [review]
Hi @Nahush26 
Right now, your code requires lot of improvement. It should follow the code design of chipnet. For the method, there shuld only be chipnet.py. move common parts to parent classes if needed. For resnet, I think you can now use the resnet script written by Rishabh to create the model. For vgg, you would need to use a more generic script.

File: trailmet/algorithms/prune/network_slimming_prune.py (hunk 2 above)
Comment (Nahush26) [reply trailmet/algorithms/prune/network_slimming_prune.py]: Sir I have removed the hard-coded parameters from the code.

File: trailmet/algorithms/prune/network_slimming_prune.py (hunk 4 above)
Comment (Nahush26) [reply trailmet/algorithms/prune/network_slimming_prune.py]: Sir I have made some changes could you please check once.

**Pull request closed without merging**